table/block_based/block_based_table_builder.cc

```diff
@@ -567,9 +571,9 @@ struct BlockBasedTableBuilder::ParallelCompressionRep {
   // Number of blocks under compression and not appended yet.
   std::atomic<uint64_t> blocks_inflight;
   // Current compression ratio, maintained by BGWorkWriteRawBlock.
-  double curr_compression_ratio;
+  std::atomic<double> curr_compression_ratio;
   // Estimated SST file size.
-  uint64_t estimated_file_size;
+  std::atomic<uint64_t> estimated_file_size;
```

Comment:
`curr_compression_ratio` will only be updated in writer. `estimated_file_size` can be updated from multiple places, but there's we don't need to be accurate when reading it, so it's kind of a blind write.